rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java

```diff
@@ -259,6 +259,28 @@ void test_getTransactionType_pegin_retiring_fed() {
         Assertions.assertEquals(PegTxType.PEGIN, pegTxType);
     }
 
+    @Test
+    void test_getTransactionType_pegin_retired_fed() {
+        // Arrange
+        when(provider.getLastRetiredFederationP2SHScript()).thenReturn(Optional.ofNullable(retiredFed.getP2SHScript()));
+        Wallet liveFederationWallet = new BridgeBtcWallet(context, Collections.singletonList(activeFederation));
+
+        BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTransaction.addOutput(Coin.COIN, retiredFed.getAddress());
+
+        // Act
+        PegTxType pegTxType = PegUtils.getTransactionTypeUsingPegoutIndex(
```

Comment:
I wouldn't test `getTransactionTypeUsingPegoutIndex` methos, that should be probably be private right? All tests should go through `getTransactionType` method using different conditions to test using the index or not